pandas/core/frame.py

```diff
@@ -4848,6 +4848,13 @@ def apply(self, func, axis=0, broadcast=False, raw=False, reduce=None,
             while guessing, exceptions raised by func will be ignored). If
             reduce is True a Series will always be returned, and if False a
             DataFrame will always be returned.
+        result_type : 'infer' or None
+            These only act when axis=1 {columns}
+            * infer : list-like results will be turned into columns
+            * None : list-like results will be returned as a list
+                     in a single column
```

Comment:
I think we should explain in more detail what the different result types do (eg the fact that Series output is also inferred in the default case is not mentioned)